RDMA/cxgb3: Mark rdma device with CXIO_ERROR_FATAL when removing.

From: Steve Wise <swise@opengridcomputing.com>

If iw_cxgb3 is called to remove itself from the low level driver, then
mark the rdma device with CXIO_ERROR_FATAL since the device below us is
going away.  Otherwise, we can get stuck in a deadlock as ULPs try and
deallocate objects (like MRs, QPs, etc).

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
---

 drivers/infiniband/hw/cxgb3/iwch.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)


diff --git a/drivers/infiniband/hw/cxgb3/iwch.c b/drivers/infiniband/hw/cxgb3/iwch.c
index 6a3214a..1ce6509 100644
--- a/drivers/infiniband/hw/cxgb3/iwch.c
+++ b/drivers/infiniband/hw/cxgb3/iwch.c
@@ -188,6 +188,7 @@ static void close_rnic_dev(struct t3cdev *tdev)
 	mutex_lock(&dev_mutex);
 	list_for_each_entry_safe(dev, tmp, &dev_list, entry) {
 		if (dev->rdev.t3cdev_p == tdev) {
+			dev->rdev.flags = CXIO_ERROR_FATAL;
 			cancel_delayed_work_sync(&dev->db_drop_task);
 			list_del(&dev->entry);
 			iwch_unregister_device(dev);
